File: src/main/java/com/nixvision/support/terminal/platform/app/domains/core/api/telegram/requests/UpdateRequest.java
package com.nixvision.support.terminal.platform.app.domains.core.api.telegram.requests;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;

@Data
@ToString
@Accessors(chain = true)
public class UpdateRequest {

    @JsonProperty("offset")
    private Integer offset;

    @JsonProperty("limit")
    private Integer limit;

    @JsonProperty("timeout")
    private Integer timeout;

}

File: src/main/kotlin/io/github/support/terminal/application/domains/integrations/telegram/requests/UpdateRequest.kt
package io.github.support.terminal.application.domains.integrations.telegram.requests

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

private const val RESPONSE_MESSAGES_LIMIT = 100

@JsonIgnoreProperties(ignoreUnknown = true)
data class UpdateRequest(
        @JsonProperty("offset") val offset: Int,
        @JsonProperty("limit") val limit: Int = RESPONSE_MESSAGES_LIMIT,
        @JsonProperty("timeout") val timeout: Int? = null
)